Model/Sync/Reset/Catalog.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace Yotpo\Core\Model\Sync\Reset;
+
+use Magento\Framework\App\Cache\TypeListInterface;
+use Magento\Framework\App\ResourceConnection;
+use Yotpo\Core\Model\Config;
+use Yotpo\Core\Model\Sync\Data\Main as SyncDataMain;
+use Yotpo\Core\Model\Config as CoreConfig;
+
+class Catalog extends Main
+{
+    const PRODUCT_SYNC_TABLE = 'yotpo_product_sync';
+    const CATEGORY_SYNC_TABLE = 'yotpo_category_sync';
+    const CRONJOB_CODES = ['yotpo_cron_core_category_sync','yotpo_cron_core_products_sync'];
+    const YOTPO_ENTITY_NAME = 'catalog';
+
+    /**
+     * @var SyncDataMain
+     */
+    protected $syncDataMain;
+
+    /**
+     * @param ResourceConnection $resourceConnection
+     * @param SyncDataMain $syncDataMain
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        Config $config,
+        TypeListInterface $cacheTypeList,
+        SyncDataMain $syncDataMain
+    ) {
+        parent::__construct(
+            $resourceConnection,
+            $config,
+            $cacheTypeList
+        );
+        $this->syncDataMain = $syncDataMain;
+    }
+
+    /**
+     * @return string
+     */
+    public function getYotpoEntityName()
+    {
+        return self::YOTPO_ENTITY_NAME;
+    }
+
+    /**
+     * @param int $storeId
+     * @param boolean $skipSyncTables
+     * @return void
+     * @throws \Zend_Db_Statement_Exception
+     */
+    public function resetSync($storeId, $skipSyncTables = false)
+    {
+        parent::resetSync($storeId);
+        $this->resetCatalogSyncAttributes($storeId);
+        $this->setResetInProgressConfig($storeId, '0');
+    }
+
+    /**
+     * @return array <string>
+     */
+    protected function getTableResourceNames()
+    {
+        return [self::PRODUCT_SYNC_TABLE, self::CATEGORY_SYNC_TABLE];
+    }
+
+    /**
+     * @return array <string>
+     */
+    protected function getCronJobCodes()
+    {
+        return self::CRONJOB_CODES;
+    }
+
+    /**
+     * @param int $storeId
+     * @return void
+     */
+    private function resetCatalogSyncAttributes($storeId)
+    {
+        $dataSet = [
+            [
+                'table_name' => 'catalog_category_entity_int',
+                'attribute_code' => CoreConfig::CATEGORY_SYNC_ATTR_CODE
+            ],
+            [
+                'table_name' => 'catalog_product_entity_int',
+                'attribute_code' => CoreConfig::CATALOG_SYNC_ATTR_CODE
+            ],
+        ];
```

Comment:
New `yotpo_collections_products_sync` should also be taken into consideration, but this doesn't have an attribute code.

Reply:
We will clear the entries from yotpo_collections_products_sync as part of deleteAllFromTables() method. This should be enough I think. We don't need to add anything in the above area becasue category product association is not managed by any attributes

Reply:
Oh I got confused, you're right.
Either way, I see you've added clearing `yotpo_collections_products_sync` in the fix commit.

Reply:
@LiorGingi ,

Hope this comment is closed or is there anything else should we need to do ?